src/webgl-utils/parameter-access.js

```diff
@@ -253,10 +268,27 @@ export function glGetParameter(gl, key) {
 // considered a very slow operation, to be used only if/when a context already manipulated
 // by external code needs to be synchronized for the first time
 // @return {Object} - a newly created map, with values keyed by GL parameters
-export function glCopyParameters(gl) {
+export function getParameters(gl, parameters) {
+  // default to querying all parameters
+  parameters = parameters || GL_PARAMETER_DEFAULTS;
+  // support both arrays of parameters and objects (keys represent parameters)
+  const parameterKeys = Array.isArray(parameters) ? parameters : Object.keys(parameters);
+
   const state = {};
-  for (const key in GL_PARAMETER_DEFAULTS) {
-    state[key] = glGetParameter(gl, key);
+  for (const key of parameterKeys) {
+    state[key] = getParameter(gl, key);
   }
   return state;
 }
+
+export function getDefaultParameters(gl) {
+  // TODO - Query GL.VIEWPORT and GL.SCISSOR_BOX since these are dynamic
+  return Object.assign({}, GL_PARAMETER_DEFAULTS, {
+    // [GL.VIEWPORT]: gl.constructor.prototype.getParameter.call(gl, GL.VIEWPORT),
```

Comment:
When commenting out code, good to add an additional comment explaining why we leave it in (rather than cut it).